Disambiguate Codestart and Extension codestarts in the doc (#13735)
quarkusio/quarkus: independent-projects/tools/codestarts/README.adoc

== Description

> "Extension Codestarts" is the name we gave to our Quarkus extension quickstart code generation system.
> Extension codestarts provide a personalized Quarkus getting started experience and really show the Quarkus breadth.

Reviewer: ```suggestion
"Extension Codestarts" is the name we gave to our Quarkus extension quickstart code generation system. It aims to provide a personalized getting started experience with Quarkus.
```

jbang cli@quarkusio create
----

> To disambiguate, we decided that, Codestart is the tool (the code generator system), a codestart is an item used by the tool, an Extension codestart is example code for an extension.

Reviewer: This sentence is confusing. 

Codestart with an initial upper case is the system and all lower case is an item used by the tool? It would be nice if that is explained more clearly here

Author: @gastaldi wdyt?
```suggestion
We decided that:

* "Codestart(s)" is the tool (the code generator system).
* A "codestart" is a directory following a specific struture containing a `codestart.yml` and templates. It is used by the tool in combinations with other codestarts to generate a project.
* An "Extension codestart" is a specific codestart providing example code for a Quarkus extension.
```

Reviewer: It reads better, but I think the main issue here is that I'm afraid the word `codestart` is being overused 😃

Author: yeah I agree, but we couldn't settle on better with @maxandersen.. 

Problem is that I named everything codestarts thinking it would only be for Quarkus extensions and now it's too late 😂

. A `codestart.yml` file
. Optionally some templates which following a common structure and naming conventions

== Where are the Quarkus Extension Codestarts located

* Quarkus core codestarts dir: https://github.com/quarkusio/quarkus/tree/master/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core
* Extension codestarts directory: https://github.com/quarkusio/quarkus/tree/master/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples

> == Writing an Extension codestart

Reviewer: ```suggestion
== Writing an Extension Codestart
```
